app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -15,6 +15,23 @@ def failed_deployments(restart_count = 5)
     current_pods.values.select { |h| h[:last_state_terminated] && h.fetch(:container_restarts, 0) > restart_count }.collect { |h| h[:label_name] }
   end
 
+  def sync_deployment_settings
+    return unless podified?
+    return unless Settings.server.worker_monitor.enforce_resource_constraints
+
+    podified_miq_workers.each do |worker|
```

Comment:
Note, we do active `podified_miq_workers` because we want active deployments.  Looping through all the worker classes would result in classes without deployments.  The downside is if there's a change to the configuration, this could mean we do the same worker class multiple times where the first detects the change in `deployment_resource_constraints_changed? => true` and the remaining workers with that class would return `deployment_resource_constraints_changed? => false`.  

I'm leaning towards tacking `worker.worker_deployment_name` in this loop and skip ones we already did.  Does this make sense?

Reply:
Fixed.  We now check if the worker_deployment_name has already been checked.